dcs/action.py

```diff
@@ -1787,6 +1787,166 @@ def dict(self):
+        d = super().dict()
+        d["file"] = self.file_res_key.key
+        d["seconds"] = self.seconds
+        d["clearview"] = self.clearview
+        d["start_delay"] = self.start_delay
+        d["horzAlignment"] = self.horz_alignment
+        d["vertAlignment"] = self.vert_alignment
+        d["size_units"] = self.size_units
+        d["size"] = self.size
+        return d
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self.__dict__ == other.__dict__
+        return False
+
+
+class PictureToAll(PictureAction):
+    predicate = "a_out_picture"
+
+    def __init__(self, file_res_key: ResourceKey, seconds: int,
+                 clearview: bool, start_delay: int,
+                 horz_alignment: str, vert_alignment: str,
+                 size: int, size_units: str):
+        super().__init__(predicate=self.predicate, file_res_key=file_res_key,
+                         seconds=seconds, clearview=clearview,
+                         start_delay=start_delay, horz_alignment=horz_alignment,
+                         vert_alignment=horz_alignment, size=size, size_units=size_units)
+        self.params = [self.file_res_key, seconds, clearview, start_delay,
+                       horz_alignment, vert_alignment, size, size_units]
+
+    @classmethod
+    def create_from_dict(cls, d, mission):
+        return cls(ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+                   d["horzAlignment"], d["vertAlignment"], d["size"], d["size_units"])
+
+    def dict(self):
+        return super().dict()
+
+
+class PictureToCoalition(PictureAction):
+    predicate = "a_out_picture_s"
+
+    def __init__(self, coalition: str, file_res_key: ResourceKey, seconds: int,
+                 clearview: bool, start_delay: int,
+                 horz_alignment: str, vert_alignment: str,
+                 size: int, size_units: str):
+        super().__init__(predicate=self.predicate, file_res_key=file_res_key,
+                         seconds=seconds, clearview=clearview,
+                         start_delay=start_delay, horz_alignment=horz_alignment,
+                         vert_alignment=horz_alignment, size=size, size_units=size_units)
+        self.coalition = coalition
+        self.params = [self.coalition, self.file_res_key, seconds, clearview,
+                       start_delay, horz_alignment, vert_alignment, size, size_units]
+
+    @classmethod
+    def create_from_dict(cls, d, mission):
+        return cls(d["coalitionlist"], ResourceKey(d["file"]), d["seconds"], d["clearview"], d["start_delay"],
+                   d["horzAlignment"], d["vertAlignment"], d["size"], d["size_units"])
+
+    def dict(self):
+        d = super().dict()
+        d["coalitionlist"] = self.coalition
+        return d
+
+
+class PictureToCountry(PictureAction):
+    predicate = "a_out_picture_c"
+
+    def __init__(self, country: int, file_res_key: ResourceKey, seconds: int,
```

Comment:
Two options here:

1. Use `Country` instead of `int`
2. Rename it `country_id`

1 is a bit nicer on the caller, but 2 might be better for testing? Up to you.

Reply:
Yeah, Python's a bit of an asshole about this. Up at the top of the file you should do:

```python
from __future__ import annotations

# rest of the normal imports
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from .country import Country
```

It's a bit funky but that is what Python expects us to do for type-only circular imports.